### PR TITLE
fix(VTimePicker): increase clock click accuracy

### LINKED ISSUE #7299: [Bug Report] Timepicker's minutes become difficult to select when using allowed-minutes prop.
### Versions and Environment
**Vuetify:** 1.5.14
**Vue:** 2.6.10
**Browsers:** Chrome 74.0.3729.157
**OS:** Mac OS 10.14.4, iOS, Android, Mac OSX

### Steps to reproduce
1. Select an hour
2. Select a minute

### Expected Behavior
Clicking or tapping between available minutes should select the closest available one.

### Actual Behavior
If you don't perfectly click/tap a minute 'circle', nothing happens.

### Reproduction Link
<a href="https://codepen.io/wallslide/pen/pmKjVE" target="_blank">https://codepen.io/wallslide/pen/pmKjVE</a>

### PR DESCRIPTION
**Pull request by jacekkarczmarczyk**
## Motivation and Context
fixes #7299

## How Has This Been Tested?
unit, playground

## Markup:

<details>

```vue
<template>
  <v-container>
    <v-time-picker
      :allowed-minutes="v => v % 30 === 0"
      :allowed-hours="v => v % 3 === 0"
    />
    <v-time-picker />
  </v-container>
</template>
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VTimePicker/__tests__/VTimePickerClock.spec.ts [hunk 1]
```diff
@@ -223,23 +233,63 @@ describe('VTimePickerClock.js', () => {
     expect(angle({ x: 2, y: 0 })).toBe(135)
   })
 
-  it('should not emit input event when clicked disabled value (#5897)', () => {
+  it.only('should calculate position from angle', () => {
```
Comment (nekosaur) [inline packages/vuetify/src/components/VTimePicker/__tests__/VTimePickerClock.spec.ts]: you forgot to remove only

File: packages/vuetify/src/components/VTimePicker/VTimePickerClock.ts [hunk 2]
```diff
@@ -197,26 +197,36 @@ export default mixins<options &
       const coords = { x: clientX - left, y: top - clientY }
       const handAngle = Math.round(this.angle(center, coords) - this.rotate + 360) % 360
       const insideClick = this.double && this.euclidean(center, coords) < (innerWidth + innerWidth * this.innerRadiusScale) / 4
+
+      for (let i = 0; i < Math.ceil(15 / this.degreesPerUnit); i++) {
```
Comment (johnleider) [inline packages/vuetify/src/components/VTimePicker/VTimePickerClock.ts]: Can `Math.ceil(15 / this.degreesPerUnit)` be assigned outside of the loop?

File: packages/vuetify/src/components/VTimePicker/VTimePickerClock.ts [hunk 3]
```diff
@@ -197,26 +197,36 @@ export default mixins<options &
...
 
       // Necessary to fix edge case when selecting left part of the value(s) at 12 o'clock
-      let newValue: number
-      if (handAngle >= (360 - this.degreesPerUnit / 2)) {
-        newValue = insideClick ? this.max - this.roundCount + 1 : this.min
+      if (angle >= (360 - this.degreesPerUnit / 2)) {
+        return insideClick ? this.max - this.roundCount + 1 : this.min
       } else {
-        newValue = value
+        return value
```
Comment (johnleider) [inline packages/vuetify/src/components/VTimePicker/VTimePickerClock.ts]: image

File: packages/vuetify/src/components/VTimePicker/VTimePickerClock.ts [hunk 4]
```diff
@@ -197,26 +197,36 @@ export default mixins<options &
...
-          this.valueOnMouseDown = newValue
-        }
-        this.valueOnMouseUp = newValue
-        this.update(newValue)
+    },
+    setMouseDownValue (value: number) {
+      if (this.valueOnMouseDown === null) {
+        this.valueOnMouseDown = value
       }
+      this.valueOnMouseUp = value
```
Comment (johnleider) [inline packages/vuetify/src/components/VTimePicker/VTimePickerClock.ts]: As outlined in the guide above, should have a blank line after the close of the conditional. While these lines are not returning, I think we should apply this here as well. If you agree, I'll update the Coding Guidelines to reflect this standard.

Comment (jacekkarczmarczyk) [reply packages/vuetify/src/components/VTimePicker/VTimePickerClock.ts]: > I'll update the Coding Guidelines to reflect this standard
 
👍

Comment (johnleider) [reply packages/vuetify/src/components/VTimePicker/VTimePickerClock.ts]: updated

Comment (johnleider) [inline packages/vuetify/src/components/VTimePicker/VTimePickerClock.ts]:
```suggestion

      this.valueOnMouseUp = value
```

Comment (jacekkarczmarczyk) [reply packages/vuetify/src/components/VTimePicker/VTimePickerClock.ts]: It's already there

**Pull request merged**